src/types/sort.ts

```diff
@@ -29,3 +29,17 @@ export interface DealerSortParams {
   seed?: number
   previewId?: number
 }
+
+export enum LeadSortTypeParams {
+  NEWEST = "NEWEST",
+}
+
+export enum LeadSortOrderParams {
+  ASC = "ASC",
+  DESC = "DESC",
+}
+
+export interface LeadSortParams {
```

Comment:
Observation: we keep creating sort param types that look very similar. Would it make sense to leverage a generic? Something along the lines of:

```typescript
interface SortParams<SortType> {
  sortOrder?: SortOrder
  sortType?: SortType
}
```

We can still export concrete types from the package if it's needed or let consumers build them on the fly

Reply:
Yes, that makes sense. I will make the changes.

Reply:
I've made generic type just as suggested with a slight change as we have alternate `SortTypes` (ListingSortTypeParams|LeadSortTypeParams) so it takes a type variable:
```
interface SortParams<T> {
  sortOrder?: SortOrder
  sortType?: T
}
```
Also made fail-safe/fallback export `SortParams as ListingSortParams`. This new `ListingSortParams` now needs sortType implied and as such will need to make that change where is used. Or maybe I should still keep exiting `ListingSortParams` that does not require `sortType` ?